evalml/data_checks/id_columns_data_check.py

```diff
@@ -175,9 +175,27 @@ def validate(self, X, y=None):
         ]  # columns whose name is "id"
         id_cols = {col: 0.95 for col in cols_named_id}
 
-        X = X.ww.select(include=["Integer", "Categorical"])
+        X_double = X.ww.select(include=["Double"])
 
-        check_all_unique = X.nunique() == len(X)
+        # Temporary solution for baton logical types mapping integers to doubles in woodwork logical types.
+        # Will be removed when resolved.
+        check_all_unique = X_double.nunique() == len(X_double)
```

Comment:
To reduce the re-used code, I wonder if we can do either a for loop:
```python
for dtypes in [['Double'], [Integer', 'Categorical']]:
   # logic here
```
or to separate out this into a helper method, where we can pass in the `Double` and `Integer/Categorical` arguments. I know this is just a stopgap fix for now as we wait, but might be nice to not have repetitive code and reuse parts when we can.